<a href="https://colab.research.google.com/github/anapmdm/template_portfolio/blob/master/Introdu%C3%A7%C3%A3o_ao_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Introdução ao Pandas**
1º: Vamos importar o *pandas* sempre **as pd**
Através do Pandas é possivel importar dados com formato *.csv e xls**, fica mais fácil extrair informações como média, mediana, valores máximos e mínimos e visualizar gráficos.


In [1]:
#Importação da Biblioteca
import pandas as pd

**Importando arquivo .csv**
Neste exemplo o arquivo importado está disponivel no site: [Yahoo Finance](https://https://finance.yahoo.com/) será os dados da ação ordinária do Banco do Brasil (BBAS3) que estão na minha pasta do github.

In [3]:
#Importando arquivo .csv para o pandas (arquivo do raw no github)
df = pd.read_csv("https://raw.githubusercontent.com/anapmdm/BBAS3/main/BBAS3.SA.csv?token=ASRA47HHYEB76KIB67G3DT3AET4FM")

**Componentes básicos do Pandas**
São séries e data frame.
O data Frame é um conjunto de séries (ou uma planilha no excel e cada série uma coluna).



*  Type() mostra qual o tipo de váriavel


In [4]:
#verificar qual o tipo da variavel df
type(df)

pandas.core.frame.DataFrame

In [5]:
#verificar o tipo de coluna de df
type(df['Date'])

pandas.core.series.Series

In [7]:
#verificando o tamanho do dataframe
df.shape

(247, 7)

Isso mostra que o arquivo importado possui 247 linhas e 7 colunas!

In [8]:
#verificando os 5 primeiras entradas do conjunto
df.head()

Date       Open       High  ...      Close  Adj Close      Volume
0  2020-02-11  49.919998  51.650002  ...  51.320000  49.331635  18096500.0
1  2020-02-12  51.869999  52.360001  ...  51.619999  49.620007  16617100.0
2  2020-02-13  51.349998  51.540001  ...  50.820000  48.851002  13806300.0
3  2020-02-14  50.880001  50.959999  ...  49.849998  47.918587  10022700.0
4  2020-02-17  50.080002  50.240002  ...  49.099998  47.197647   6749000.0

[5 rows x 7 columns]

In [9]:
#verificando as 5 ultimas entradas do conjunto
df.tail()

Date       Open       High  ...      Close  Adj Close      Volume
242  2021-02-04  34.599998  34.720001  ...  34.189999  34.189999  10074200.0
243  2021-02-05  34.279999  34.480000  ...  34.000000  34.000000   8052300.0
244  2021-02-08  33.939999  34.250000  ...  33.869999  33.869999  11651600.0
245  2021-02-09  33.770000  34.500000  ...  34.279999  34.279999  15307000.0
246  2021-02-10  34.279999  34.480000  ...  33.830002  33.830002  10459600.0

[5 rows x 7 columns]

In [10]:
#verificar o nome das colunas
df.columns

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume'], dtype='object')

**Tipos de Variaveis**
* Para dados como receita, despesas ou lucros : **float**.
* Para anos: **int**
* Para datas (11-02-2021): **datetime**


In [11]:
#Conhecendo os tipos 
df.dtypes
#incorreto Date apresentar como object

Date          object
Open         float64
High         float64
Low          float64
Close        float64
Adj Close    float64
Volume       float64
dtype: object

In [12]:
#conhecendo os tipos e verificando se há dados em branco 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 247 entries, 0 to 246
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       247 non-null    object 
 1   Open       246 non-null    float64
 2   High       246 non-null    float64
 3   Low        246 non-null    float64
 4   Close      246 non-null    float64
 5   Adj Close  246 non-null    float64
 6   Volume     246 non-null    float64
dtypes: float64(6), object(1)
memory usage: 13.6+ KB


**Selecionando Colunas**
Caso o nome da coluna seja simples basta digitar df**.**nomedacoluna,se não é necessário colocar df ["nome da coluna"]

In [15]:
#selecionando a coluna com "", não precisa ponto
df["High"]

0      51.650002
1      52.360001
2      51.540001
3      50.959999
4      50.240002
         ...    
242    34.720001
243    34.480000
244    34.250000
245    34.500000
246    34.480000
Name: High, Length: 247, dtype: float64

In [17]:
#selecionando a coluna de nome simples diretamente, atentar para o ponto
df.High

0      51.650002
1      52.360001
2      51.540001
3      50.959999
4      50.240002
         ...    
242    34.720001
243    34.480000
244    34.250000
245    34.500000
246    34.480000
Name: High, Length: 247, dtype: float64

**Calculando a Média**
Para calculo de média usa-se .mean.

In [18]:
#calculando a média da coluna High
df.High.mean()

34.3945936097561

**Trabalhando com Datas**


In [19]:
#verificando o tipo de variavel da coluna Date
type("Date")
#Apresenta como string ou seja texto

str

É necessário modificar o tipo da coluna Date transformando ela de object (string) para datetime.

In [21]:
#antes
df.Date

0      2020-02-11
1      2020-02-12
2      2020-02-13
3      2020-02-14
4      2020-02-17
          ...    
242    2021-02-04
243    2021-02-05
244    2021-02-08
245    2021-02-09
246    2021-02-10
Name: Date, Length: 247, dtype: object

In [30]:
#transformando para datetime apresentando o formato que eu quero ler
df.Date = pd.to_datetime(df.Date, format= '%Y-%m-%d')


In [29]:
#verificando
df.Date

0     2020-02-11
1     2020-02-12
2     2020-02-13
3     2020-02-14
4     2020-02-17
         ...    
242   2021-02-04
243   2021-02-05
244   2021-02-08
245   2021-02-09
246   2021-02-10
Name: Date, Length: 247, dtype: datetime64[ns]

Agora é possivel pegar informações como ano, mÊs e dia separadamente.


In [32]:
#buscando o ano **dt é abreviação de datetime**
df.Date.dt.year

0      2020
1      2020
2      2020
3      2020
4      2020
       ... 
242    2021
243    2021
244    2021
245    2021
246    2021
Name: Date, Length: 247, dtype: int64

In [33]:
#mês
df.Date.dt.month

0      2
1      2
2      2
3      2
4      2
      ..
242    2
243    2
244    2
245    2
246    2
Name: Date, Length: 247, dtype: int64

In [34]:
#dia
df.Date.dt.day

0      11
1      12
2      13
3      14
4      17
       ..
242     4
243     5
244     8
245     9
246    10
Name: Date, Length: 247, dtype: int64

In [35]:
#dia da semana
df.Date.dt.dayofweek

0      1
1      2
2      3
3      4
4      0
      ..
242    3
243    4
244    0
245    1
246    2
Name: Date, Length: 247, dtype: int64

**Para casos em que as datas estão apresentadas em formatos diferentes verificar os links:**
(https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html?highlight=to_datetime#pandas.to_datetime)
(https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior)